tests/python/pants_test/backend/native/tasks/test_c_compile.py

```diff
@@ -38,10 +54,29 @@ def create_simple_c_library(self, **kwargs):
                             sources=['test.h', 'test.c'],
                             **kwargs)
 
+  def test_header_only_noop_with_alternate_header_extension(self):
+    alternate_extension = '.asdf'
+    c = self.create_header_only_alternate_c_library(alternate_extension)
+    context = self.prepare_context_for_compile(target_roots=[c], options={
+      'c-compile-settings': {
+        'header_file_extensions': [alternate_extension],
+      },
+    })
+
+    # Test that the task runs without error if provided a header-only library.
+    c_compile = self.create_task(context)
+    c_compile.execute()
+
+    object_files_product = context.products.get(ObjectFiles)
+    object_files_for_target = self._retrieve_single_product_at_target_base(object_files_product, c)
+    # Test that no object files were produced.
```

Comment:
Thanks Danny - this is a much better test!

Reply:
I think so too! I swear to you that I prevaricated for at least an hour about adding tests based on matching info logs and moved forward only to unblock myself from that particular indecision. I am planning to take your comment on the other PR and make a (concise!) followup issue to fix this in the integration tests from that PR.